packages/jest-util/src/deepCyclicCopy.ts

```diff
@@ -84,3 +100,69 @@ function deepCyclicCopyArray<T>(
 
   return newArray;
 }
+
+function deepCyclicCopyMap<T>(
+  map: Map<any, any>,
+  options: DeepCyclicCopyOptions,
+  cycles: WeakMap<any, any>,
+): T {
+  const newMap = new Map();
+
+  cycles.set(map, newMap);
+
+  map.forEach((value, key) => {
+    if (options.blacklist && options.blacklist.has(key)) return;
+    newMap.set(
+      key,
+      deepCyclicCopy(
+        value,
+        {blacklist: EMPTY, keepPrototype: options.keepPrototype},
+        cycles,
+      ),
+    );
+  });
+
+  return newMap as any;
+}
+
+function deepCyclicCopySet<T>(
+  set: Set<T>,
+  options: DeepCyclicCopyOptions,
+  cycles: WeakMap<any, any>,
+): T {
+  const newSet = new Set();
+
+  cycles.set(set, newSet);
+
+  set.forEach(value => {
+    newSet.add(
+      deepCyclicCopy(
+        value,
+        {blacklist: EMPTY, keepPrototype: options.keepPrototype},
+        cycles,
+      ),
+    );
+  });
+
+  return newSet as any;
+}
+
+function deepCyclicCopyRegExp<T>(regExp: RegExp): T {
+  const newRegExp = new RegExp(regExp.source, regExp.flags);
+  newRegExp.lastIndex = regExp.lastIndex;
+  return newRegExp as any;
+}
+
+function isNumberArray(value: any): boolean {
+  return [
+    Int8Array,
+    Uint8Array,
+    Uint8ClampedArray,
+    Int16Array,
+    Uint16Array,
+    Int32Array,
+    Uint32Array,
+    Float32Array,
+    Float64Array,
+  ].includes(value.constructor);
```

Comment:
```suggestion
  ].includes(value && value.constructor);
```

or add an invariant check on `value` above